src/main/java/org/breedinginsight/services/parsers/trait/TraitFileParser.java

```diff
@@ -115,30 +126,56 @@
                     .formula(parseExcelValueAsString(record, TraitFileColumns.METHOD_FORMULA))
                     .build();
 
-            // TODO: throw if not valid datatype
             DataType dataType = null;
+            String dataTypeString = parseExcelValueAsString(record, TraitFileColumns.SCALE_CLASS);
+            if (dataTypeString == null) {
+                throw new ParsingException("Invalid scale class value");
```

Comment:
Should this message be "Missing scale class value"?

Reply:
I did have it as missing in the service check but in the parser I was using invalid to signify that the value is not one of a valid set of values. I can change it to missing for consistency.

Reply:
Pushed changes.